libraries/Storage/Microsoft.Bot.Builder.Azure.Blobs/BlobsStorage.cs

```diff
@@ -77,7 +78,7 @@ public BlobsStorage(string dataConnectionString, string containerName, StorageTr
 
             _jsonSerializer = jsonSerializer ?? JsonSerializer.Create(new JsonSerializerSettings
                                                 {
-                                                    TypeNameHandling = TypeNameHandling.All,
+                                                    TypeNameHandling = TypeNameHandling.None,
```

Comment:
We probably need update the class and constructor documentation for this class since it is really important that devs know that this will have consequences on how state is managed. We can even tell them that TypeNameHandling.All should be **only** used for backward compat and completely discouraged. Let's chat about how we make this more prominent, we could throw warning to the logger, make the constructor that let's override typehandling obsolete so they get a compiler warning or things like that. 
We should make sure the documentation is consistent among the OOTB IStorage implementations